### PR TITLE
[enhancement](thirdparty) upgrade thirdparty libs

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

protobuf 3.15.0 -> 21.11
glog 0.4.0 -> 0.6.0
lz4 1.9.3 -> 1.9.4
curl 7.79.0 -> 8.2.1
brpc 1.4.0 -> 1.6.0
zstd 1.5.2 -> 1.5.5
arrow 7.0.0 -> 13.0.0
abseil 20220623.1 -> 20230125.3
orc 1.7.2 -> 1.9.0
jemalloc for arrow 5.2.1 -> 5.3.0
xsimd 7.0.0 -> 13.0.0
opentelemetry-proto 0.19.0 -> 1.0.0
opentelemetry 1.8.3 -> 1.10.0

new:
c-ares  -> 1.19.1
grpc -> 1.54.3

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.71 seconds
 stream load tsv:          553 seconds loaded 74807831229 Bytes, about 129 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162299264 Bytes

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.7 seconds
 stream load tsv:          537 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17162073019 Bytes

File: thirdparty/vars.sh [hunk 1]
```diff
@@ -197,10 +198,10 @@ LEVELDB_SOURCE=leveldb-1.23
 LEVELDB_MD5SUM="afbde776fb8760312009963f09a586c7"
 
 # brpc
-BRPC_DOWNLOAD="https://github.com/apache/brpc/archive/refs/tags/1.4.0.tar.gz"
-BRPC_NAME="brpc-1.4.0.tar.gz"
-BRPC_SOURCE="brpc-1.4.0"
-BRPC_MD5SUM="6af9d50822c33a3abc56a1ec0af0e0bc"
+BRPC_DOWNLOAD="https://github.com/apache/brpc/archive/refs/tags/1.6.0.tar.gz"
+BRPC_NAME="1.6.0.tar.gz"
```
Comment (adonis0147) [inline thirdparty/vars.sh:202]:
```suggestion
BRPC_NAME="brpc-1.6.0.tar.gz"
```

File: thirdparty/vars.sh [hunk 2]
```diff
@@ -398,11 +412,11 @@ BENCHMARK_SOURCE=benchmark-1.8.0
...
 
 # xsimd
-# for arrow-7.0.0, if arrow upgrade, this version may also need to be changed
-XSIMD_DOWNLOAD="https://github.com/xtensor-stack/xsimd/archive/aeec9c872c8b475dedd7781336710f2dd2666cb2.tar.gz"
-XSIMD_NAME=xsimd-aeec9c872c8b475dedd7781336710f2dd2666cb2.tar.gz
-XSIMD_SOURCE=xsimd-aeec9c872c8b475dedd7781336710f2dd2666cb2
-XSIMD_MD5SUM="d024855f71c0a2837a6918c0f8f66245"
+# for arrow-13.0.0, if arrow upgrade, this version may also need to be changed
+XSIMD_DOWNLOAD="https://github.com/xtensor-stack/xsimd/archive/refs/tags/9.0.1.tar.gz"
+XSIMD_NAME="9.0.1.tar.gz"
```
Comment (adonis0147) [inline thirdparty/vars.sh:417]:
```suggestion
XSIMD_NAME="xsimd-9.0.1.tar.gz"
```

File: thirdparty/vars.sh [hunk 3]
```diff
@@ -417,16 +431,16 @@ NLOHMANN_JSON_SOURCE=json-3.10.1
 NLOHMANN_JSON_MD5SUM="7b369d567afc0dffdcf5800fd9abb836"
 
 # opentelemetry-proto
-OPENTELEMETRY_PROTO_DOWNLOAD="https://github.com/open-telemetry/opentelemetry-proto/archive/refs/tags/v0.19.0.tar.gz"
-OPENTELEMETRY_PROTO_NAME=opentelemetry-proto-0.19.0.tar.gz
-OPENTELEMETRY_PROTO_SOURCE=opentelemetry-proto-0.19.0
-OPENTELEMETRY_PROTO_MD5SUM="8a5fb1ba721341994a5e7cae8b38bcc6"
+OPENTELEMETRY_PROTO_DOWNLOAD="https://github.com/open-telemetry/opentelemetry-proto/archive/refs/tags/v1.0.0.tar.gz"
+OPENTELEMETRY_PROTO_NAME=v1.0.0.tar.gz
```
Comment (adonis0147) [inline thirdparty/vars.sh:435]:
```suggestion
OPENTELEMETRY_PROTO_NAME="opentelemetry-proto-v1.0.0.tar.gz"
```

**Review by adonis0147 — CHANGES_REQUESTED** [review]

**Comment by xinyiZzz** [comment]
run buildall

File: thirdparty/vars.sh (hunk 1 above)
Comment (xinyiZzz) [reply thirdparty/vars.sh]: fix

File: thirdparty/vars.sh (hunk 2 above)
Comment (xinyiZzz) [reply thirdparty/vars.sh:417]: fixed

File: thirdparty/vars.sh (hunk 3 above)
Comment (xinyiZzz) [reply thirdparty/vars.sh]: fixed

**Comment by xinyiZzz** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.97 seconds
 stream load tsv:          554 seconds loaded 74807831229 Bytes, about 128 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162085456 Bytes

**Review by adonis0147 — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**